### ISSUE TITLE
Suggestion: use an options dictionary for functions with lots of args

### ISSUE THREAD

**Issue by zolkis** [webmachinelearning/webnn#90]
A syntax-only question/suggestion. There may be reasons for the current design (disclaimer: I am long C) but wouldn't the JavaScript code look better with shorter signature containing the operands and the rest would be named in a dictionary?

For instance, now there is 

```javascript 
partial interface NeuralNetworkContext {
  Operand gemm(Operand a, Operand b, optional Operand c, 
               optional float alpha = 1.0, optional float beta = 1.0, 
               optional boolean aTranspose = false, optional boolean bTranspose = false);
};
```
and it could be also written (following the pattern used in many other web APIs) as

```javascript
partial interface NeuralNetworkContext {
  Operand gemm(Operand a, Operand b, optional Operand c, optional GemmOptions options);
};

dictionary GemmOptions {
   float alpha = 1.0;
   float beta = 1.0; 
   boolean aTranspose = false; 
   boolean bTranspose = false;
};
```
while the code using defaults would stay the same, and code using non-defaults would look more clear:

```javascript
let r = nn.gemm(a, b, c);

let q = nn.gemm(a, b, c, { alpha: 1.1, beta: 1.1, aTranspose:  true });
```
as opposed to 
```javascript
let q = nn.gemm(a, b, c, 1.1, 1.1, true);
```
Maybe not so big improvement for `gemm`, but bigger opportunities for the likes of `gru` ops, improving clarity of the function semantics and code readability as well.

I am aware this doesn't add much value to counter the arguments for the original design, if there is more to that than to make it look like a C API :).

**Comment by wchao1115** [comment]
The suggestion looks reasonable. The only concern I have would be that if we apply this style of API to `gemm` we would probably need to deliver the same treatment across all other API for consistency, which would likely introduce many more *options* types. Secondly, it might be a bit ambiguous to differentiate what property should be in the options and what not.

**Comment by anssiko** [comment]
I discussed this proposal with the editors and there is agreement this proposal has merit.

In order to make timely progress, and given no call next week, I mandate the editors to work with @zolkis to craft a PR for the group to review.

@zolkis introduced this proposal on 1 Oct call and we recorded no concerns at that time.

**Comment by wchao1115** [comment]
Fixed with PR #112.
